### PR TITLE
chore(DDI-975):adding missing examples of button component.

### PR DESCRIPTION
**Pull request by Radha-Alberta**
Added examples for variant, title and disabled props.

### CODE REVIEW

File: libs/docs/src/components/common/Button.stories.mdx [hunk 1]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+      variant="destructive"
+      onClick={() => console.log('clicked')}
+    >
+      Destructive Tertiary
+    </GoAButton>
+  </GoAButtonGroup>
+</Story>
+
+<Tabs>
+  <Tab label="Web Component" hidden={true}>
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:343]: We don't need a tab for Web Component, I don't think we're doing that anywhere else.

File: libs/docs/src/components/common/Button.stories.mdx [hunk 2]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+          console.log('clicked');
+        });
+      `}
+      lang="js"
+    />
+  </Tab>
+  <Tab label="Angular">
+    <CodeSnippet
+      code={`
+        <goa-button variant="destructive" (_click)="onClick()" type="primary">Destructive Primary</goa-button>
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:364]: You can remove the _click event, it's not required.

File: libs/docs/src/components/common/Button.stories.mdx [hunk 3]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+            console.log('clicked');}
+        }
+      `}
+      lang="ts"
+    />
+  </Tab>
+  <Tab label="React">
+    <CodeSnippet
+      code={`
+        <GoAButton variant="destructive" type="primary" onClick={() => console.log('clicked')}>
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:383]: Can remove the onClick, it's not required

File: libs/docs/src/components/common/Button.stories.mdx [hunk 4]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
   </Tab>
 </Tabs>
 
+#### variant
+
+There are two variant types: Normal (default) and Destructive.
+
+#### Normal variant button
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:237]: I would argue we don't need to show anything for normal variant, every other button on this stories page is already a normal variant as that's the default, there's nothing really to show.

File: libs/docs/src/components/common/Button.stories.mdx [hunk 5]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
   </Tab>
 </Tabs>
 
+#### variant
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:233]: Probably should uppercase `variant`

File: libs/docs/src/components/common/Button.stories.mdx [hunk 6]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+        <GoAButton variant="destructive" type="tertiary" onClick={() => console.log('clicked')}>
+          Destructive Tertiary
+        </GoAButton>
+    `}
+      lang="tsx"
+    />
+  </Tab>
+</Tabs>
+
+#### title
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:398]: Uppercase

File: libs/docs/src/components/common/Button.stories.mdx [hunk 7]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+
+<Story name="Title">
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:400]: I'm looking at storybook, and there's not a `title` property under properties. Am I going crazy? I wrote this story, and I'm sure it was there when I wrote it, but it's not there now that I'm looking. If it's still a property that can be used, maybe add it back to properties as well.

File: libs/docs/src/components/common/Button.stories.mdx [hunk 8]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+</Tabs>
+
+#### Destructive variant button
+
+<Story name="Destructive variant">
+  <GoAButtonGroup alignment="start">
+    <GoAButton
+      type="primary"
+      variant="destructive"
+      onClick={() => console.log('clicked')}
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:321]: You can remove all the onClicks, they aren't required

File: libs/docs/src/components/common/Button.stories.mdx [hunk 9]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+</Tabs>
+
+#### title
+
+<Story name="Title">
+  <GoAButtonGroup alignment="start">
+    <GoAButton
+      type="primary"
+      title="Go to Next Page."
+      onClick={() => console.log('clicked')}
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:405]: Remove all the onClicks, not required

File: libs/docs/src/components/common/Button.stories.mdx [hunk 10]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+      title="Click here to open the link."
+      onClick={() => console.log('clicked')}
+    >
+      Click Me
+    </GoAButton>
+  </GoAButtonGroup>
+</Story>
+
+<Tabs>
+  <Tab label="Web Component" hidden={true}>
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:427]: Don't need a tab for web components

File: libs/docs/src/components/common/Button.stories.mdx [hunk 11]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+          console.log('clicked');
+        });
+      `}
+      lang="js"
+    />
+  </Tab>
+  <Tab label="Angular">
+    <CodeSnippet
+      code={`
+        <goa-button type="primary" (_click)="onClick()" title="Go to Next Page.">Next</goa-button>
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:448]: Can remove all the onClicks, not required

File: libs/docs/src/components/common/Button.stories.mdx [hunk 12]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+        <goa-button type="submit" (_click)="onClick()" title="Submit the form.">Submit</goa-button>
+        <goa-button type="tertiary" (_click)="onClick()" title="Click here to open the link.">Click Me</goa-button>
+      `}
+      lang="html"
+    />
+    <CodeSnippet
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:454]: Don't need this CodeSnippet

File: libs/docs/src/components/common/Button.stories.mdx [hunk 13]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+            console.log('clicked');}
+        }
+      `}
+      lang="ts"
+    />
+  </Tab>
+  <Tab label="React">
+    <CodeSnippet
+      code={`
+        <GoAButton type="primary" onClick={() => console.log('clicked')} title="Go to Next Page.">
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:467]: onClicks not required, can remove them

File: libs/docs/src/components/common/Button.stories.mdx [hunk 14]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+    <GoAButton type="primary" disabled="true">
+      Primary
+    </GoAButton>
+    <GoAButton type="submit" disabled="true">
+      Submit
+    </GoAButton>
+    <GoAButton type="secondary" disabled="true">
+      Secondary
+    </GoAButton>
+  </GoAButtonGroup>
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:495]: Maybe add a tertiary button as well?

File: libs/docs/src/components/common/Button.stories.mdx [hunk 15]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+</Story>
+
+<Tabs>
+  <Tab label="Web Component" hidden={true}>
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:499]: Don't need a tab for web components

File: libs/docs/src/components/common/Button.stories.mdx [hunk 16]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+        <goa-button type="secondary" disabled="true">Secondary</goa-button>
+      `}
+      lang="html"
+    />
+  </Tab>
+  <Tab label="React">
+    <CodeSnippet
+      code={`
+        <GoAButton type="primary" disabled={true}>
+        Primary
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Button.stories.mdx:523]: Indent these things inside a button

File: libs/docs/src/components/common/Button.stories.mdx [hunk 17]
```diff
@@ -230,6 +230,310 @@ There are two button sizes: Normal (default) and Compact.
...
+  <Tab label="Angular">
+    <CodeSnippet
+      code={`
+        <goa-button variant="destructive" (_click)="onClick()" type="primary">Destructive Primary</goa-button>
+        <goa-button variant="destructive" (_click)="onClick()" type="secondary">Destructive Secondary</goa-button>
+        <goa-button variant="destructive" (_click)="onClick()" type="tertiary">Destructive Tertiary</goa-button>
+      `}
+      lang="html"
+    />
+    <CodeSnippet
```
[the remark this answers is not in the document]
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the CodeSnippet

File: libs/docs/src/components/common/Button.stories.mdx (hunk 2 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the click event

File: libs/docs/src/components/common/Button.stories.mdx (hunk 3 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the click event

File: libs/docs/src/components/common/Button.stories.mdx (hunk 4 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the normal variant section

File: libs/docs/src/components/common/Button.stories.mdx (hunk 5 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Done

File: libs/docs/src/components/common/Button.stories.mdx (hunk 6 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Done

File: libs/docs/src/components/common/Button.stories.mdx (hunk 8 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the click event

File: libs/docs/src/components/common/Button.stories.mdx (hunk 9 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the click event

File: libs/docs/src/components/common/Button.stories.mdx (hunk 10 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the web component tab

File: libs/docs/src/components/common/Button.stories.mdx (hunk 11 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the click event

File: libs/docs/src/components/common/Button.stories.mdx (hunk 12 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the CodeSnippet

File: libs/docs/src/components/common/Button.stories.mdx (hunk 13 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the click event

File: libs/docs/src/components/common/Button.stories.mdx (hunk 14 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx:273]: Added the tertiary button

File: libs/docs/src/components/common/Button.stories.mdx (hunk 15 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the web component tab

File: libs/docs/src/components/common/Button.stories.mdx (hunk 16 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Done.

File: libs/docs/src/components/common/Button.stories.mdx (hunk 1 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx]: Removed the web component tab

File: libs/docs/src/components/common/Button.stories.mdx (hunk 7 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Button.stories.mdx:220]: I could see it in the code and also my local env. 
button_properties

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**